lib/client.ts

```diff
@@ -4,17 +4,17 @@ import { getModels } from "./graphql";
 import { GraphQLClient } from "./graphql/client";
 import { ClientOpts } from "./types";
 
-export class Client {
+export class Client<T extends Partial<ClientOpts> = ClientOpts> {
   constructor(
-    opts: ClientOpts,
+    opts: T,
     baseUrl = opts.baseUrl || KONTIST_API_BASE_URL,
     baseSubscriptionUrl = opts.baseSubscriptionUrl || KONTIST_SUBSCRIPTION_API_BASE_URL,
-    public auth = new Auth(baseUrl, opts),
+    public auth: T extends { clientId?: string | undefined } ? Auth : undefined = opts.clientId ? new Auth(baseUrl, opts as ClientOpts) : undefined as any,
```

Comment:
`opts.clientId` can be `""`. I'll update the code to check for it

Reply:
```
    public auth: T extends { clientId?: string | undefined } ? Auth : undefined ...
```
We want to test if `T` extends the type that *has* this `clientId`. So it should be
```
    public auth: T extends { clientId: string } ? Auth : undefined ...
```

Reply:
So what this means is:
`auth` should be of type `Auth` if the type of `opts` (which is `T`) contains the `clientId`. If that is not the case the type of `auth` should be `undefined`.

Reply:
I understand that part, but doing so will result in eslint error `Object is possibly 'undefined'.`

For example
```
tests/graphql/client.spec.ts:22:7 - error TS2532: Object is possibly 'undefined'.

22       client.auth.tokenManager.setToken("dummy-access-token");
         ~~~~~~~~~~~
````

Reply:
The issue is because of the check `opts.clientId ?`, it would return false if `opts.clientId` is empty string.

Reply:
@johannespfeiffer can you help to double check please? Thanks.